### PR TITLE
Fix task.resolved_action callbacks

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY

Noticed while looking at the related fix to not use this attribute for module_defaults in https://github.com/ansible/ansible/pull/81909.

Not sure if there is a better way to set this - it feels a bit redundant to load the plugin context additional times.

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

**Comment by s-hertel** [comment]
@nitzmahone Not sure if you have any thoughts on this, but after thinking about this in conjunction with module_defaults, I'm wondering if the `resolved_action` attribute is just rather unhelpful for collections that use a common action plugin.

For example, say vyos_mod1 and vyos_mod2 are redirected to the action plugin vyos. To prevent module_defaults for vyos_mod1 clobbering the default args for vyos_mod2 or vice versa, they can add the `action_plugin` directive for modules in the runtime.yml.

If these tasks run while using a callback that displays the `resolved_action`, they will all be displayed as the vyos action, rather than the potentially-more-helpful module name. This is only sort of similar to the [module_defaults issue](https://github.com/ansible/ansible/issues/77059) (basically unusable feature vs bad readability). Could I adjust the resolved_action to reflect the resolved module in this case, or is that overloading the task attr/the module metadata magic?

**Comment by s-hertel** [comment]
I didn't actually realize this also fixes https://github.com/ansible/ansible/issues/81905, but it does so I added tests and a changelog.

File: test/units/playbook/test_block.py [hunk 1]
```diff
@@ -63,16 +63,16 @@ def test_load_block_with_tasks(self):
...
         self.assertIsInstance(b.block[0], Task)
 
     def test_deserialize(self):
         ds = dict(
-            block=[dict(action='block')],
-            rescue=[dict(action='rescue')],
-            always=[dict(action='always')],
+            block=[dict(action='ping')],
+            rescue=[dict(action='ping')],
+            always=[dict(action='ping')],
```
Comment (bcoca) [inline test/units/playbook/test_block.py]: i'm not sure why these are changed, doesn't making them all 'ping' defeat the purpose of testing the block segments?

Comment (s-hertel) [reply test/units/playbook/test_block.py]: Oh, good catch. Reverting my changes to the tests show the plugin loader isn't initialized

```
=================================== FAILURES ===================================
___________________ TestLoadListOfTasks.test_unknown_action ____________________
[gw19] linux -- Python 3.12.2 /usr/bin/python3.12

self = <units.playbook.test_helpers.TestLoadListOfTasks testMethod=test_unknown_action>

    def test_unknown_action(self):
        action_name = 'foo_test_unknown_action'
        ds = [{'action': action_name}]
>       res = helpers.load_list_of_tasks(ds, play=self.mock_play,
                                         variable_manager=self.mock_variable_manager, loader=self.fake_loader)

test/units/playbook/test_helpers.py:128: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
lib/ansible/playbook/helpers.py:297: in load_list_of_tasks
    t = Task.load(task_ds, block=block, role=role, task_include=task_include, variable_manager=variable_manager, loader=loader)
lib/ansible/playbook/task.py:134: in load
    return t.load_data(data, variable_manager=variable_manager, loader=loader)
lib/ansible/playbook/base.py:160: in load_data
    ds = self.preprocess_data(ds)
lib/ansible/playbook/task.py:202: in preprocess_data
    (action, args, delegate_to) = args_parser.parse()
lib/ansible/parsing/mod_args.py:309: in parse
    context = _get_action_context(action, self._collection_list)
lib/ansible/parsing/mod_args.py:57: in _get_action_context
    module_context = module_loader.find_plugin_with_context(action_or_module, collection_list=collection_list)
lib/ansible/plugins/loader.py:591: in find_plugin_with_context
    result = self._resolve_plugin_step(name, mod_type, ignore_deprecated, check_aliases, collection_list, plugin_load_context=plugin_load_context)
lib/ansible/plugins/loader.py:687: in _resolve_plugin_step
    return self._find_plugin_legacy(name, plugin_load_context, ignore_deprecated, check_aliases, suffix)
lib/ansible/plugins/loader.py:792: in _find_plugin_legacy
    return self._find_fq_plugin(fq_name=candidate_fqcr, extension=suffix, plugin_load_context=plugin_load_context, ignore_deprecated=ignore_deprecated)
lib/ansible/plugins/loader.py:481: in _find_fq_plugin
    routing_metadata = self._query_collection_routing_meta(acr, plugin_type, extension=extension)
lib/ansible/plugins/loader.py:439: in _query_collection_routing_meta
    collection_pkg = import_module(acr.n_python_collection_package_name)
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'ansible_collections', import_ = <function _gcd_import at 0x7fc36c8540e0>

>   ???
E   ModuleNotFoundError: No module named 'ansible_collections'

<frozen importlib._bootstrap>:1324: ModuleNotFoundError
```

The difference is the task FA `action` is now exercising the plugin loader (whereas before `action`/`local_action` were excluded), and I must have just assumed the helpers actually needed the plugins to exist. I reverted the renaming and fixed more correctly by initializing the plugin loader.

**Review by bcoca — APPROVED** [review]

**Pull request merged**